### DISCUSSION TITLE
File Upload Compact?

### DISCUSSION THREAD

**Discussion by mtanco** [h2oai/wave#1409] [Q&A]
Are you required to add your own button for the File Upload component when `compact=True`?

**Comment by mturoci** [comment]
Currently yes. Is there a scenario where this UX may not be the best?

**Reply by mtanco** [reply]
No no but the https://wave.h2o.ai/docs/widgets/form/file_upload/ page could make this more clear. I thought the compact option looked nice but then thought maybe it was broken or something because I couldn't "get my data in". It then occurred to me I just needed my own button but I think it's a little off that making it compact takes functionality away - maybe just adding that note on that page would help.

**Reply by mturoci** [reply]
Agree, makes sense.

**Reply by mturoci** [reply]
Done in https://github.com/h2oai/wave/commit/295cc61bd5eb2626fa84bc47346373d3f6e28efc
